parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -66,6 +66,9 @@ class ParquetWriter<T> implements FileAppender<T>, Closeable {
   private boolean closed;
   private ParquetFileWriter writer;
   private int rowGroupOrdinal;
+  private long currentBufferSize = 0;
```

Comment:
Since this tracks mostly what is being added in memory directly on each `add`, I'd suggest renaming to `currentRawBufferedSize` or something alike.